lib/ecmarkdown.js

```diff
@@ -1,11 +1,357 @@
 'use strict';
-var parser = require('./generated-parser.js');
+
 var escapeHtml = require('escape-html');
+var beautify = require('./beautify.js');
 
-exports.list = function (ecmarkdown) {
-    return parser.parse(ecmarkdown.trim() + '\n', { startRule: 'list' });
+exports.document = function (ecmarkdown) {
+  return beautify(emitDocument(parse(ecmarkdown)));
 };
 
 exports.fragment = function (ecmarkdown) {
-    return parser.parse(escapeHtml(ecmarkdown.trim()), { startRule: 'fragment' });
+  return beautify(emitFragment(parse(escapeHtml(ecmarkdown))));
 };
+
+
+var lexemes = {
+  star: /\*/,
+  underscore: /_/,
+  tick: /`[^`\n]*`/,
```

Comment:
I've unified the parsing of all formats and disallowed nesting (if we want to allow nesting later it's a trivial change).

All formats are now handled uniformly with respect to how they can be started and ended (eg. `foo| bar` is not a valid start and `foo |bar` is not a valid end) and whether spaces are allowed.